File: 13_reversingaudio.m
% Load the audio file
[audio, fs] = audioread('trumpet');

% Reverse the signal
reversed_audio = flipud(audio);

% Create a time vector
t = (0:length(audio)-1) / fs;

% Plot the original and reversed signals
subplot(2, 1, 1)
plot(t, audio)
title('Original Signal')
xlabel('Time (s)')
ylabel('Amplitude')

subplot(2, 1, 2)
plot(t, reversed_audio)
title('Reversed Signal')
xlabel('Time (s)')
ylabel('Amplitude')

File: 13_reversingaudio.py
import librosa
import matplotlib.pyplot as plt
import numpy as np

# Load the audio file
audio_path = librosa.example('trumpet')
signal, sr = librosa.load(audio_path, sr=None)

# Reverse the signal
reversed_signal = np.flipud(signal)

# Create a time vector
t = librosa.times_like(signal, sr=sr)

# Plot the original and reversed signals
plt.subplot(2, 1, 1)
plt.plot(t, signal)
plt.title('Original Signal')
plt.xlabel('Time (s)')
plt.ylabel('Amplitude')

plt.subplot(2, 1, 2)
plt.plot(t, reversed_signal)
plt.title('Reversed Signal')
plt.xlabel('Time (s)')
plt.ylabel('Amplitude')

plt.tight_layout()
plt.show()

